client/components/ImageService.jsx

```diff
@@ -0,0 +1,128 @@
+import React from 'react';
+import axios from 'axios';
+import _ from 'lodash';
+
+import Modal from './Modal.jsx';
+import LightBox from './LightBox.jsx';
+
+import style from "../styles/style.css";
+
+class ImageService extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      didFetch: false,
+      openModal: false,
+      allImagesLoaded: false,
+      imageCount: 0,
+      curImageIndex: 0,
+      images: []
+    }
+    this.fetchNewImages = this.fetchNewImages.bind(this);
+    this.onImageLoad = this.onImageLoad.bind(this);
+    this.openModal = this.openModal.bind(this);
+    this.closeModal = this.closeModal.bind(this);
+  }
+
+  componentDidMount() {
+    console.log(this.props.locationId);
+    this.fetchNewImages(this.props.locationId);
+  }
+
+  componentDidUpdate(prevProps) {
```

Comment:
@leealbert95  What does componetDidUpdate do here?

Reply:
This is a React lifecycle that fires when the component updates as a result of state/prop change. I had to make sure that whenever the prop (location id) is changed, that the component makes the API call to fetch the images for the new location id.